bin/ch/ch.cpp

```diff
@@ -199,31 +199,41 @@ static void CALLBACK PromiseContinuationCallback(JsValueRef task, void *callback
     messageQueue->InsertSorted(msg);
 }
 
-HRESULT RunScript(const char* fileName, LPCWSTR fileContents, BYTE *bcBuffer, char *fullPath)
+static bool CHAKRA_CALLBACK DummyJsSerializedScriptLoadUtf8Source(_In_ JsSourceContext sourceContext, _Outptr_result_z_ const char** scriptBuffer)
+{
+    *scriptBuffer = reinterpret_cast<const char*>(sourceContext);
+    return true;
+}
+
+static void CHAKRA_CALLBACK DummyJsSerializedScriptUnload(_In_ JsSourceContext sourceContext)
+{
+    free(reinterpret_cast<void*>(sourceContext));
```

Comment:
Can you add some comments on the lifetime management here? I understand what is going on but it'd be more clear with comments here to others (like, why we're calling free here when the Load callback counterpart is not doing any malloc etc).